assets/js/modules/thank-with-google/datastore/publications.test.js

```diff
@@ -184,47 +190,53 @@ describe( 'modules/thank-with-google publications', () => {
 						},
 					] );
 
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.setPublicationID( 'test-publication-id-2' );
+
 				const publication = registry
 					.select( MODULES_THANK_WITH_GOOGLE )
 					.getCurrentPublication();
 
-				expect( publication ).toEqual( PUBLICATIONS[ 0 ] );
+				expect( publication ).toEqual( {
+					// eslint-disable-next-line sitekit/acronym-case
+					publicationId: 'test-publication-id-2',
+					displayName: 'Test publication title',
+					verifiedDomains: [ 'https://example.com' ],
+					paymentOptions: {
+						virtualGifts: true,
+					},
+				} );
 			} );
 
-			it( 'returns the publication if the publicationId is not set and the state is set to ACTIVE', () => {
-				const publicationsWithActiveState = [
-					{
-						displayName: 'Test publication title',
-						verifiedDomains: [ 'https://example.com' ],
-						paymentOptions: {
-							virtualGifts: true,
-						},
-					},
-					// The following publication doesn't have the publicationID.
-					// However, it has the state set to ACTIVE.
-					{
-						displayName: 'Test publication another title',
-						verifiedDomains: [ 'https://example.com' ],
-						paymentOptions: {
-							virtualGifts: true,
+			it( 'returns the publication if the publicationID is not set and the state is set to ACTIVE', () => {
+				registry
+					.dispatch( MODULES_THANK_WITH_GOOGLE )
+					.receiveGetPublications( [
+						...PUBLICATIONS,
+						{
+							// eslint-disable-next-line sitekit/acronym-case
+							publicationId: 'test-publication-id-2',
+							displayName: 'Test publication title',
+							verifiedDomains: [ 'https://example.com' ],
+							paymentOptions: {
+								virtualGifts: true,
+							},
 						},
-						state: 'ACTIVE',
-					},
-				];
+					] );
```

Comment:
For example, this test is about a publication with the `state: ACTIVE` but this isn't visible anywhere in the test but "hidden" in `PUBLICATIONS` as implicit state. Let's make this explicit in the test.